nulib/dc-api-v2
# Handle DC Chat Feedback Submission

### Summary

Handle DC chat feedback submission. 
- Saves response to S3
  - Files can be overwritten in the case of a initial 👍 / 👎  followed by a more detailed response
- Sends notification/email to subscribers of topic

### Steps to Test

- Fetch the branch and clean/build
- Deploy a development branch: [Deploy Development Branch](https://github.com/nulib/dc-api-v2?tab=readme-ov-file#deploying-a-development-branch)
- Go into the AWS SNS staging console and subscribe to the topic that was created, which should be named `{your-dev-env}-chat-feedback)
  - Look for the subscription confirmation in your email inbox and follow instuctions
- Visit the `https://your-dev-branch-name.rdc-staging.library.northwestern.edu/api/v2/chat-feedback` route (You can use the Postan POST chat feedback endpoint (screenshot below))
- You should receive validation errors for invalid JSON
- You should receive a success message/200 from the API when the form is submitted 
- You should receive an email notifying you of the response (screenshot below)
- Find the `{your-dev-env}-chat-feedback` bucket and look for the response with a key of `{sentiment}/{ref}.json` 


<img width="1062" alt="Screenshot 2024-07-01 at 9 14 25 AM" src="https://github.com/nulib/dc-api-v2/assets/6372022/adfc87b6-52d3-4fed-86b1-954342e7a1ab">

<img width="812" alt="Screenshot 2024-07-01 at 9 14 45 AM" src="https://github.com/nulib/dc-api-v2/assets/6372022/cdcb929f-b4b2-4008-ac52-e4537b750fe3">
